src/runtime/str.cpp

```diff
@@ -1632,11 +1632,11 @@ extern "C" Box* strNonzero(BoxedString* self) {
     return boxBool(self->size() != 0);
 }
 
-extern "C" Box* strNew(BoxedClass* cls, Box* obj) {
+template <ExceptionStyle S> Box* strNew(BoxedClass* cls, Box* obj) {
```

Comment:
Minor, but can you add a `noexcept(S == CAPI)` right before the opening brace?  This will mean that the CAPI version will get a "noexcept" qualifier and is just an extra protection against unhandled exceptions.